### DISCUSSION TITLE
Suggestions about switch network

### DISCUSSION THREAD

**Discussion by begmuhommet** [rainbow-me/rainbowkit#606] [Feedback]
Hi. Thank you for building this great library. It is so easy to use.

My Problem:

When I connect my wallet to my dapp Rainbow Kit force me to switch network.
Problem is I have 5 defined networks. Rainbow kit force me switch to first one.
I think it is not convinient. I won't switch network if I not doing any transactions. 

Suggestions:
 1. Not switch network when connecting wallet. Just can show "Wrong network" instead of network. (better)
 2. Not switch network when my network have in defined networks.

**Comment by sweetmantech** [comment]
+1 on this.

I wrote about this exact experience on Twitter here: https://twitter.com/sweetman_eth/status/1566066327713890305

**Comment by Ryan-Ouyang** [comment]
+1 to this!

If your app supports multiple chains, it's an unneeded extra step to prompt the switch to Mainnet (or whatever the first chain is) as soon as they connect. It's especially annoying when I connect trying to do something on Polygon, but then it switches to Mainnet and I have to go to my wallet to switch back myself.

**Comment by kopy-kat** [comment]
You can just define all wallets as viable networks in your config and that should solve the issue. For example you could do this:

```
import {
  allChains
} from 'wagmi'

const { chains } = configureChains(
  allChains,
  [
    alchemyProvider({ apiKey: process.env.ALCHEMY_ID }),
    publicProvider(),
  ]
);

**Reply by begmuhommet** [reply]
@kopy-kat Hi. Yes it will solve 1 problem and give me another problem. 
 - Users can not changing chain when they connect wallet. (solved)
 - In chain modal users will see all chains. (problem)
![Screenshot_2](https://user-images.githubusercontent.com/29680223/200203729-4bba1180-e3c6-4363-a6ff-e217f28d3b8e.png)

**Reply by begmuhommet** [reply]
I am supporting just 3 chains. Ethereum, BSC and EktaChain. So I need show just 3 chains and change chain when they not in that 3 chains. Thank you for your interest.
I am not using @rainbowkit anymore. I just using wagmi instead.

**Reply by kopy-kat** [reply]
In the case of only needing some chains, you can manually define them by importing them from wagmi and putting them in an array as the first input to the configureChains function. If you want to support any chain but not show the chain modal, you can use the code above and on the ConnectButton set chains=“none” which will hide the option to change chains in your dapp.

**Reply by begmuhommet** [reply]
@kopy-kat I will try that solution. Thank you very much !

**Reply by samishal1998** [reply]
It doesn't solve the issue.
The issue is that when you're supporting multiple chains prompting the user to change network to the first network in the array is an unnecessary extra step for the user, and sometimes annoying especially if the user is already on the network he intends to connect with.

Like for example; Just minutes ago My Wallet is set on polygon and I wanted to connect to my App, I was prompted to switch to ethereum (1st network in the list) as a part of the connection process and if I reject to switch network then the connection fails, finally I have to switch back to polygon when I initially was on Polygon

**Reply by haytham1996** [reply]
`
            
                 import { polygon } from "wagmi/chains";
                              const config = getDefaultConfig({
                              appName: "AppName",
                               projectId: "ID",
                               chains: [polygon],
  
                                });`
you can put the chains you want in the chains array and this will solve your problem .
